core/src/data_sources/data_source.rs

```diff
@@ -594,23 +596,99 @@ impl DataSource {
             utils::now() - now
         ));
 
+        // Ordered results with offset, stirng and vector
+        let mut texthash2embedding: HashMap<String, EmbedderVector> = HashMap::new();
+        let mut page_offset: Option<PointId> = None;
+        loop {
+            let scroll_results = qdrant_client
+                .scroll(&ScrollPoints {
+                    collection_name: self.qdrant_collection(),
+                    with_vectors: Some(true.into()),
+                    limit: Some(1000),
+                    offset: page_offset,
+                    filter: Some(qdrant::Filter {
+                        must_not: vec![],
+                        should: vec![],
+                        must: vec![qdrant::FieldCondition {
+                            key: "document_id_hash".to_string(),
+                            r#match: Some(qdrant::Match {
+                                match_value: Some(qdrant::r#match::MatchValue::Text(
+                                    document_id_hash.clone(),
+                                )),
+                            }),
+                            ..Default::default()
+                        }
+                        .into()],
+                    }),
+                    ..Default::default()
+                })
+                .await?;
+
+            scroll_results
+                .result
+                .iter()
+                .for_each(|result| match result.payload.get("text") {
+                    Some(t) => match t.kind {
+                        Some(qdrant::value::Kind::StringValue(ref chunk_text)) => {
+                            match result.vectors.clone() {
+                                Some(v) => match v.vectors_options {
+                                    Some(VectorsOptions::Vector(v)) => {
+                                        let mut hasher = blake3::Hasher::new();
+                                        hasher.update(chunk_text.as_bytes());
+                                        let text_only_hash =
```

Comment:
probably useless to hash it here since it's going to be hashed my the hashmap. Won't change the memory complexity that would actually increase the memory foot print.

Reply:
`test_hash` is shorter and equally clear?